pkg/webhook/pod/tikv_creater.go

```diff
@@ -84,3 +89,37 @@ func (pc *PodAdmissionControl) admitCreateTiKVPod(pod *core.Pod, tc *v1alpha1.Ti
 
 	return util.ARSuccess()
 }
+
+// This is method is to make compatible between old pdapi version and 4.0 pdapi version.
+// To get more detail, see: https://github.com/pingcap/tidb-operator/pull/1831
+func filterLeaderEvictScheduler(evictLeaderSchedulers []string, pdClient pdapi.PDClient) (sets.String, error) {
+	schedulerIds := sets.String{}
+	if len(evictLeaderSchedulers) == 1 && evictLeaderSchedulers[0] == evictSchedulerLeader {
+		c, err := pdClient.GetConfig()
+		if err != nil {
+			return schedulerIds, err
+		}
+		if c.Schedule != nil {
+			if c.Schedule.SchedulersPayload == nil {
+				c.Schedule.SchedulersPayload = map[string]string{}
+			}
+			v, ok := c.Schedule.SchedulersPayload[evictSchedulerLeader]
```

Comment:
Could you please give an example of the `c`? or any doc link about this new behavior?
I did not find the `c.Schedule.SchedulersPayload` in my cluster.

Reply:
It's the new payload in pd 4.0

Reply:
Here is an example for the payload response：
```bash
"balance-hot-region-scheduler": "null",
"balance-leader-scheduler": "{\"name\":\"balance-leader-scheduler\",\"ranges\":[{\"start-key\":\"\",\"end-key\":\"\"}]}",
"balance-region-scheduler": "{\"name\":\"balance-region-scheduler\",\"ranges\":[{\"start-key\":\"\",\"end-key\":\"\"}]}","evict-leader-scheduler": "{\"store-id-ranges\":{\"4\":[{\"start-key\":\"\",\"end-key\":\"\"}]}}",
"label-scheduler": "{\"name\":\"label-scheduler\",\"ranges\":[{\"start-key\":\"\",\"end-key\":\"\"}]}"

```